src/test/codegen/intrinsics/mask.rs

```diff
@@ -0,0 +1,7 @@
+#![crate_type = "lib"]
+#![feature(core_intrinsics)]
+
+pub fn mask_ptr(ptr: *const u8, mask: usize) -> *const u8 {
+    // CHECK: llvm.ptrmask
```

Comment:
Given that https://github.com/rust-lang/rust/pull/96946#discussion_r870708503 suggests that this isn't the full name, could this be a more specific check?  Something like this, maybe:
```rust
// CHECK-LABEL: @mask_ptr
// CHECK-SAME: [[WORD:i[0-9]+]] %mask
#[no_mangle]
pub fn mask_ptr(ptr: *const u8, mask: usize) -> *const u8 {
    // CHECK: call
    // CHECK-SAME: @llvm.ptrmask.p0i8.[[WORD]](
    core::intrinsics::ptr_mask(ptr, mask)
}
```

Reply:
So the llvm-ir looks like this:
```llvm
; Function Attrs: nonlazybind uwtable
define i8* @mask_ptr(i8* %ptr, i64 %mask) unnamed_addr #0 {
start:
  %0 = alloca i8*, align 8
  %1 = bitcast i8* %ptr to void*
  %2 = call void* @llvm.ptrmask.p0isVoid.i64(void* %1, i64 %mask)
  %3 = bitcast i8** %0 to void**
  store void* %2, void** %3, align 8
  %4 = load i8*, i8** %0, align 8
  br label %bb1

bb1:                                              ; preds = %start
  ret i8* %4
}
```

Interestingly enough llvm seems to be casting the pointer to `void*` and only then using the intrinsic. Should I do the following?
```rust
// CHECK-LABEL: @mask_ptr
// CHECK-SAME: [[WORD:i[0-9]+]] %mask
#[no_mangle]
pub fn mask_ptr(ptr: *const u8, mask: usize) -> *const u8 {
    // CHECK: call
    // CHECK-SAME: @llvm.ptrmask.p0isVoid.[[WORD]](
    core::intrinsics::ptr_mask(ptr, mask)
}
```

Reply:
Ah, sure, if that's what the name is, that looks good.  Making it clear exactly what the real name that'll be used is -- especially when my understanding was wrong! -- is exactly what I'm hoping for with the test 🙂

Reply:
cc @nikic, actually -- should it be `@llvm.ptrmask.{{ptr|p0isVoid}}.[[WORD]](` to make sure not to fail in the [opaque pointers](https://llvm.org/docs/OpaquePointers.html) future?